tile-generation/src/main/scala/com/oculusinfo/tilegen/tiling/TilePipelines.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2014 Oculus Info Inc.
+ * http://www.oculusinfo.com/
+ *
+ * Released under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.oculusinfo.tilegen.tiling
+
+import com.oculusinfo.tilegen.tiling.TilePipelines.PipelineArgParser
+import org.apache.spark.sql.{SQLContext, SchemaRDD}
+
+/**
+ * Defines the data that is passed from stage to stage of the tile pipeline.
+ */
+case class PipelineData(sqlContext: SQLContext, srdd: SchemaRDD, tableName: Option[String] = None)
+
+/**
+ * Defines a pipeline stage that consists of an operation, a name and a list of child stages.
+ */
+case class PipelineStage(name: String, op: PipelineData => PipelineData, var children: List[PipelineStage] = List()) {
+  def addChild(child: PipelineStage): PipelineStage = {
+    children = children :+ child
+    child
+  }
+}
+
+object TilePipelines {
+
+  /**
+   * Function type that takes a map of arguments and binds them to a pipeline
+   * operation.
+   */
+  type PipelineArgParser = Map[String, String] => (PipelineData => PipelineData)
+
+  case class Empty(i: Int)
+
+  /**
+   *  Executes the pipeline via depth first tree traversal.
+   */
+  def execute(start: PipelineStage, sqlContext: SQLContext, input: Option[PipelineData] = None) = {
+    // TODO: Should run a check for cycles here (tsort?)
+    def ex(stage: PipelineStage, result: PipelineData): Unit = {
+      val stageResult = stage.op(result)
+      stage.children.foreach(ex(_, stageResult))
+    }
+
+    input match {
+      case Some(i) => ex(start, i)
+      case None =>
+        // TODO: Find correct way to create an empty schema RDD
+        val emptySchema = sqlContext.createSchemaRDD(sqlContext.sparkContext.parallelize(List(new Empty(0))))
```

Comment:
val emptySchema = sqlc.jsonRDD(sc.emptyRDD[String], new StructType(Seq()))

That should get you something with no columns at all.  I think your way gives you something with a single column.